webapp/src/Service/ImportExportService.php

```diff
@@ -83,15 +99,47 @@ public function getContestYamlData(Contest $contest, bool $includeProblems = tru
         return $data;
     }
 
+    protected function convertImportedTime(array $fields, array $data, ?string &$message = null): ?DateTimeImmutable
+    {
+        $timeValue = null;
+        $usedField = null;
+        foreach ($fields as $field) {
+            if ($timeValue) {
+                continue;
+            }
+            $timeValue = $data[$field] ?? null;
+            $usedField = $field;
+        }
+
+        if (is_string($timeValue)) {
+            $time = date_create_from_format(DateTime::ISO8601, $timeValue) ?:
+                // Make sure ISO 8601 but with the T replaced with a space also works.
+                date_create_from_format('Y-m-d H:i:sO', $timeValue);
+        } else {
+            /** @var DateTime $time */
+            $time = $timeValue;
+        }
+        if ($time === false) {
+            $message = 'Can not parse '.$usedField;
+            return null;
+        } elseif ($time) {
+            $time = $time->setTimezone(new DateTimeZone(date_default_timezone_get()));
+        }
+        return $time instanceof DateTime ? DateTimeImmutable::createFromMutable($time) : $time;
+    }
+
     public function importContestData($data, ?string &$message = null, string &$cid = null): bool
     {
         if (empty($data) || !is_array($data)) {
             $message = 'Error parsing YAML file.';
             return false;
         }
 
-        $requiredFields = [['start_time', 'start-time'], 'name', ['id', 'short-name'], 'duration'];
-        $missingFields  = [];
+        $activateTimeFields = ['activate-time', 'activation-time', 'activate_time', 'activation_time'];
+        $deactivateTimeFields = preg_filter('/^/', 'de', $activateTimeFields);
+        $startTimeFields = ['start_time', 'start-time'];
+        $requiredFields = [$startTimeFields, 'formal_name', ['id', 'name', 'short-name', 'short_name'], 'duration'];
```

Comment:
`formal_name` should not be required, we should just set our `name` to either `name` from the `contest.yaml` or if that doesn't exist the `formal_name`. So I guess it should be `['name', 'formal_name']` in there.